libosmscout-import/src/osmscout/import/GenLocationIndex.cpp

```diff
@@ -94,8 +95,26 @@ namespace osmscout {
     return false;
   }
 
-  bool LocationIndexGenerator::Region::CouldContain(const Region& region) const
+  bool LocationIndexGenerator::Region::CouldContain(const Region& region, bool strict) const
   {
+    if (strict) {
```

Comment:
Maybe name is not the best, but its not tolerant as well :). When strict is true, we check first whether the overall bounding box can contain the proposed child region. If this check fails we don't even look for intersection and reject the proposed child. As such, its an additional check which makes it harder to satisfy the condition tested by the method.